DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -102,6 +109,7 @@ internal class DataUploadWorker: DataUploadWorkerType {
                 DD.logger.debug("💡 (\(self.featureName)) No upload. Batch to upload: \(batchLabel), System conditions: \(blockersForUpload.description)")
 
                 self.delay.increase()
+                self.backgroundTaskCoordinator?.endCurrentBackgroundTaskIfActive()
```

Comment:
❓ How do we make sure `endCurrentBackgroundTaskIfActive` always called?

I think in a `defer` block would make more sense.